src/python/scripts/dx-docker

```diff
@@ -258,20 +258,43 @@ def run(args):
     to_mount = [ '/etc/host.conf', '/etc/hosts', '/etc/mtab', '/etc/networks', '/etc/passwd',
                  '/etc/group', '/etc/nsswitch.conf', '/etc/resolv.conf', '/etc/localtime',
                  '/dev/', '/sys/', '/proc/', '/run/', '/var/run/dbus/system_bus_socket']
+
+    # TODO: rename this: 'bounded' is a bit confusing and they are not always dirs
     bounded_dirs = []
     for b in to_mount:
         bounded_dirs.append('-b')
         bounded_dirs.append(b)
 
     # TODO: implement some form of nested subprocesses to avoid quoting command array if possible
+    #proot_cmd =  ["proot", '-r', '{tmpdir}/rootfs'.format(tmpdir=tmpdir)] + \
+    #             bounded_dirs + \
+    #             volume + \
+    #             ["-w", workdir] + \
+    #             env + ["/bin/sh", "-c"] + \
+    #             [" ".join(("'"+x+"'" for x in container_cmd))]
+
+
+    tempsh_file_descriptor, tempsh_file_path = tempfile.mkstemp(suffix='.sh')
+    with os.fdopen(tempsh_file_descriptor, 'w') as tmpsh:
+        lines = ['#!/bin/sh']
+        lines.extend(env[2:])
+        lines.append(subprocess.list2cmdline(container_cmd))
+        tmpsh.write('\n'.join(lines))
+
+    os.chmod(tempsh_file_path, 0o777)
```

Comment:
Is it necessary to give the script 777 permissions? Wouldn't an equivalent of 'chmod +x' suffice?

Reply:
Yes, thanks!  I'm working on testing a few modifications to make the changes cleaner.  And will probably submit them as a separate PR so it won't block the functionality. I"ll be sure to include this change as well.